tree/tree/src/InternalTreeUtils.cxx

```diff
@@ -23,6 +23,53 @@ namespace ROOT {
 namespace Internal {
 namespace TreeUtils {
 
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add information of a single friend.
+///
+/// \param[in] treeName Name of the tree.
+/// \param[in] fileNameGlob Path to the file. Refer to TChain::Add for globbing rules.
+/// \param[in] alias Alias for this friend.
+void RFriendInfo::AddFriend(const std::string &treeName, const std::string &fileNameGlob, const std::string &alias)
+{
+   fFriendNames.emplace_back(std::make_pair(treeName, alias));
+   fFriendFileNames.emplace_back(std::vector<std::string>{fileNameGlob});
+   fFriendChainSubNames.emplace_back();
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add information of a single friend.
+///
+/// \param[in] treeName Name of the tree.
+/// \param[in] fileNameGlobs Paths to the files. Refer to TChain::Add for globbing rules.
+/// \param[in] alias Alias for this friend.
+void RFriendInfo::AddFriend(const std::string &treeName, const std::vector<std::string> &fileNameGlobs,
+                            const std::string &alias)
+{
+   fFriendNames.emplace_back(std::make_pair(treeName, alias));
+   fFriendFileNames.emplace_back(fileNameGlobs);
+   fFriendChainSubNames.emplace_back(std::vector<std::string>(fileNameGlobs.size(), treeName));
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// \brief Add information of a single friend.
+///
+/// \param[in] treeAndFileNameGlobs Pairs of (treename, filename). Refer to TChain::Add for globbing rules.
+/// \param[in] alias Alias for this friend.
+void RFriendInfo::AddFriend(const std::vector<std::pair<std::string, std::string>> &treeAndFileNameGlobs,
+                            const std::string &alias)
+{
+   fFriendNames.emplace_back(std::make_pair("", alias));
+   fFriendFileNames.emplace_back();
+   fFriendChainSubNames.emplace_back();
+
+   auto fSubNamesIt = std::back_inserter(fFriendChainSubNames.back());
+   auto fNamesIt = std::back_inserter(fFriendFileNames.back());
+   for (const auto &names : treeAndFileNameGlobs) {
```

Comment:
I would suggest for:
```cpp
   for (const auto &[subNames, names] : treeAndFileNameGlobs) {
      *fSubNamesIt = subNames;
      *fNamesIt = names;
   }
```
anyway, it should do the same thing, but looks a tiny bit more readable to me.

Reply:
I believe that's a C++17 feature unfortunately

Reply:
actually, shouldn't that be the de fact standard now?